zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/SharedZkClient.java

```diff
@@ -112,4 +112,9 @@ public String create(final String path, Object datat, final List<ACL> acl,
   protected boolean isManagingZkConnection() {
     return false;
   }
+
+  // test only
+  public HelixZkClient getConnectionManager() {
+    return _connectionManager;
+  }
```

Comment:
Can you avoid adding public methods like this that shouldn't otherwise be used in production?

Reply:
Java has not way to access otherwise private members in the class. In this case, we have to expire the session inside this class. This is a common problem I did not see addressed in our code base. Unless you guys can identify another pattern, I guess we have to leave it this way.

Reply:
Agreed here, we should never add a public method to a public class just for testing.  If you really need to expose ZkClient, either change it to protected, and inherit this class in your test, such that:

class SharedZkClient {
   protected _zkClient;
}

class testSharedZkClient extends SharedZkClient {

 Protected getZkClient() {
      return _zkClient;
 }

Reply:
Another way is, if your test class is put in the same module with sharedZkClient, such as org.apache.helix.zookeeper.impl.client,  you should be able to access protected instance variables or methods from the test code.

Reply:
Here is the case:

test case is in package org.apache.helix.store.zk, here is the how it is constructed.

`    ZkSerializer serializer = new ZNRecordSerializer();
    ZkHelixPropertyStore<ZNRecord> store =
        new ZkHelixPropertyStore<>(ZK_ADDR, serializer, subRoot, subscribedPaths);`

We need the internal shared ZkClient in the store; and further the internal _ZkconnectionManager.

Note, ZkHelixPropertyStore uses ZkCacheBasedDataAccessor which lives in package org.apache.helix.manager.zk;

Note, ShareZkClient and ZkConnectionManager lives in package org.apache.helix.zookeeper.impl.client;

So the solution (in this case, but not in general) can be:

I can put the test in the org.apache.helix.zookeeper.impl.client, zookeeper-api module

I can also initialize the HelixPropertystore like this:

1/ construct SharedZkClient 

2/ construct ZkBaseDataAccessor using this sharedZkCLient.

3/ construct ZkHelixPropertyStore (using ZkBaseDataAccessor.

by making SharedZkClient's _zkConnectionManager protected, we can use it here.  

The problem is that org.apache.helix.zookeeper.impl.client is in Zookeeper-api, we need a test scope dependency to Helix-core and Helix-common module. Does this test dependency work here?